mpv.py

```diff
@@ -1036,9 +1062,17 @@ def set_loglevel(self, level):
 
     def command(self, name, *args):
         """Execute a raw command."""
-        args = [name.encode('utf-8')] + [ (arg if type(arg) is bytes else str(arg).encode('utf-8'))
-                for arg in args if arg is not None ] + [None]
-        _mpv_command(self.handle, (c_char_p*len(args))(*args))
+        args = _create_null_term_cmd_arg_array(name, args)
+        _mpv_command(self.handle, args)
+
+    def command_async(self, name, *args, callback=None):
+        """Same as mpv_command, but run the command asynchronously. Once the command ran, the callback will be invoked,
+        if provided. The first argument of the callback will be a boolean value. It will be set to True, if there was an
+        error, False else. The second argument of the callback depends on the command.
+        """
```

Comment:
> /**
     * This is mainly used for events that are replies to (asynchronous)
     * requests. It contains a status code, which is >= 0 on success, or < 0
     * on error (a mpv_error value). Usually, this will be set if an
     * asynchronous request fails.
     * Used for:
     *  MPV_EVENT_GET_PROPERTY_REPLY
     *  MPV_EVENT_SET_PROPERTY_REPLY
     *  MPV_EVENT_COMMAND_REPLY
     */
    int error;

https://github.com/mpv-player/mpv/blob/e1536193cbca6506d31b1139cd35bd575bf6e368/libmpv/client.h#L1731
So the current doc string is wrong. The question is, do we want to leak the error code to the user (making the callback return an integer), or do we just tell them, there was an error.
If the error code points to one of the errors mentioned here https://github.com/mpv-player/mpv/blob/e1536193cbca6506d31b1139cd35bd575bf6e368/libmpv/client.h#L265, we probably should change the signature of the callback altogether :see_no_evil:

Reply:
The doc in client.h is clear that this is an mpv_error. The sync ```command(...)``` method will forward this as a python exception that is raised by ErrorCode.raise_for_ec. I think passing the raw error code is ok here, a caller can always do this in their callback if they wish:
```python
def callback(error, result):
    try:
        mpv.ErrorCode.raise_for_ec(error)
        ... # handle normal case

    except MemoryError as e: # for example
        ... # handle MemoryError
    
    except Exception as e:
        ... # catch-all: handle all other exceptions
```
This looks kind of unelegant, but that's what you get with callback-based async foo. I think it's tolerable for the first version of this API.

What we could do however is to decode the MPV error ID into a human-readable string in the default callback using [_mpv_error_string](https://github.com/jaseg/python-mpv/blob/master/mpv.py#L556).

Reply:
Alright, thanks! I've updated the documentation of our `command_async` function and I also included your example code in it :smiley: 
Our default handler now also prints out the human readable version of the error code.